### PR TITLE
Xamarin iOS Support

### PR DESCRIPTION
**Pull request by rmarinho**
Since there's a couple of changes after last month pr of #280 , here's a new pr for Xamarin support #276 , the idea is to add Xamarin iOS, Mac and Android support.

- Add Xamarin iOS support for Yoga, by providing a wrapper for the native lib.
- Add Xamarin iOS Unit testing project
~~- Adds a YogaKit shared implementation in c#.~~
~~- Adds a YogaKIt iOS implementation.~~
~~- Adds Yoga/YogaKit iOS sample application~~

Facebook.Yoga.iOS.Tests.exe : 169 ms
Tests run: 114 Passed: 114 Inconclusive: 0 Failed: 0 Ignored: 0

simulator screen shot 6 jan 2017 02 05 15

### CODE REVIEW

File: csharp/Facebook.Yoga/Native.cs [hunk 1]
```diff
@@ -89,8 +89,10 @@ protected override bool ReleaseHandle()
         [DllImport(DllName, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
         public static extern void YGNodePrint(YGNodeHandle node, YogaPrintOptions options);
 
+#if !__IOS__
         [DllImport(DllName, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
         [return: MarshalAs(UnmanagedType.I1)]
+#endif
         public static extern bool YGValueIsUndefined(float value);
```
Comment (splhack) [inline csharp/Facebook.Yoga/Native.cs]: ah, we can just remove `YGValueIsUndefined` declaration from Native.cs

File: csharp/Facebook.YogaKit/YogaKit.cs [hunk 2]
```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Collections.Generic;
+using Facebook.Yoga;
+#if __IOS__
+using NativeView = UIKit.UIView;
+#endif
+
+namespace Facebook.YogaKit
+{
+	public static class YogaKit
```
Comment (splhack) [inline csharp/Facebook.YogaKit/YogaKit.cs]: Do you mind to convert tab to 4 spaces in C# source code as same as the others?

**Comment by rmarinho** [comment]
@splhack updated

**Comment by splhack** [comment]
Thank you for updating! Have you seen #322 new YogaKit PR? With my #327, I think we can just use C# YogaNode for `@property (nonatomic, readonly, retain) YGLayout *yoga`. we can remove `YogaWidth` and other methods. So maybe we could use something like

    root.Yoga.AlignItems = YogaAlign.Center;
    root.Yoga.Width = 100;
    root.Yoga.MarginTop = 10;

**Comment by rmarinho** [comment]
Seems #322 was reverted..

**Comment by emilsjolander** [comment]
@rmarinho yup, due to an internal issue. Nothing wrong with the PR itself. Working on getting it back in today or tomorrow.

**Comment by emilsjolander** [comment]
@rmarinho @splhack is this ready to be merged?

**Comment by splhack** [comment]
@emilsjolander not yet, this refers old YogaKit impl. maybe we can split this into two part, Xamarin iOS project files (like #278), and new YogaKit support.

**Comment by rmarinho** [comment]
Sure

**Comment by rmarinho** [comment]
Rebase and updated!

**Comment by splhack** [comment]
We changed a couple of things in the merge commit. Aligned directory name (`Xamarin.iOS` to `iOS`) with others (`Windows` and `Mac`), removed `AssemblyInfo`, added copyright in C# code and create a new `.gitignore`.

**Comment by rmarinho** [comment]
Ok, had a pr coming to that directory change! thanks!

**Comment by vjeux** [comment]
<3

**Pull request closed without merging**